haskell-src/exec/Chainweb/Server.hs

```diff
@@ -660,11 +632,14 @@ recentTxsHandler logger pool = liftIO $ do
           return $ (toDbTxSummary tx, contHist)
       return $ uncurry dbToApiTxSummary <$> res
 
-getTransactionCount :: LogFunctionIO Text -> P.Pool Connection -> IO (Maybe Int64)
-getTransactionCount logger pool = do
-    P.withResource pool $ \c -> do
-      runBeamPostgresDebug (logger Debug . T.pack) c $ runSelectReturningOne $ select $
-        aggregate_ (\_ -> as_ @Int64 countAll_) (all_ (_cddb_transactions database))
+getTransactionCountEstimate :: Connection -> IO Int64
+getTransactionCountEstimate c =
+    query_ c stmt
+      >>= \case
+        [Only (estimate :: Int64)] -> return estimate
+        _ -> throwIO $ userError "Could not get transaction count estimate"
+  where
+    stmt = "SELECT reltuples::bigint AS estimate FROM pg_class WHERE relname='transactions'"
```

Comment:
I've just realized that querying the `pg_class` table like this can be fragile for `chainweb-data` operators that point their instance at a database that stores other data in it. For example, if the database has a `transactions` table in a schema other than the one `chainweb-data` writes to, this query will return multiple rows even if that schema is not on the `SEARCH_PATH` of `chainweb-data` and I believe even in the case when that table isn't even accessible to `chainweb-data`. Observe:
``` sql
=> CREATE SCHEMA some_external_schema;
CREATE SCHEMA
=> CREATE TABLE some_external_schema.transactions (a INT);
CREATE TABLE
=> SELECT reltuples::bigint AS estimate FROM pg_class WHERE relname='transactions';
 estimate 
----------
 16392690
       -1
(2 rows)
``` 
A more robust solution is to select the `transactions` table by its `oid`:
``` sql
=> SELECT reltuples::bigint AS estimate FROM pg_class WHERE oid='transactions'::regclass::oid;
 estimate 
----------
 16392690
(1 row)
```
The expression `'transactions'::regclass::oid` will pick the `transactions` table using the same search method that's used when we refer to the `transactions` table in any other query.

So, here's my suggestion:
```suggestion
    stmt = "SELECT reltuples::bigint AS estimate FROM pg_class WHERE oid='transactions'::regclass::oid"
```